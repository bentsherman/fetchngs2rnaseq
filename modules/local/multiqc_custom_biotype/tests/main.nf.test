nextflow_process {

    name "Test Process MULTIQC_CUSTOM_BIOTYPE"
    script "../main.nf"
    process "MULTIQC_CUSTOM_BIOTYPE"

    test("test sarscov2 count data succesfully completes") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file(params.pipelines_testdata_base_path + 'multiqc_custom_biotype/test.featureCounts.txt', checkIfExists: true)
                    ]
                )
                input[1] = file(params.pipelines_testdata_base_path + 'multiqc_custom_biotype/biotypes_header.txt', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success},
                { assert snapshot(process.out).match()}
            )
        }

    }

}
