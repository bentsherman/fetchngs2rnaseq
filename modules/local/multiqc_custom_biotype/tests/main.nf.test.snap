{
    "test sarscov2 count data succesfully completes": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.biotype_counts_mqc.tsv:md5,d996f27aeec64370cade26717aa22e1e",
                            "test.biotype_counts_rrna_mqc.tsv:md5,405217e65a76b104b00bb41f6bf10a92"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,575b3b4c8a8a6e508c8ceeca41331b6b"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test.biotype_counts_mqc.tsv:md5,d996f27aeec64370cade26717aa22e1e",
                            "test.biotype_counts_rrna_mqc.tsv:md5,405217e65a76b104b00bb41f6bf10a92"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,575b3b4c8a8a6e508c8ceeca41331b6b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T09:38:04.260609"
    }
}